synapse/federation/federation_base.py

```diff
@@ -78,7 +79,7 @@ def _check_sigs_and_hashes(
 
         ctx = current_context()
 
-        def callback(_, pdu: EventBase):
+        async def callback(_, pdu: EventBase):
```

Comment:
I think this is the issue. A `Deferred`, doesn't know what to do with awaitables, I think we can either make this an `inlineCallbacks` call or restructure this code deeper. Nothing obvious is coming to mind at the moment though.

Reply:
I *think* I fixed that, but I still get the exact same error messages, so... I guess I didn't actually fix it.